File: exF.py
import sys

input = sys.stdin.read
print = sys.stdout.write

def find_limit():
    t = int(input().strip())  # Number of test cases
    results = []
    
    for _ in range(t):
        low, high = 0, 10**14
        balance = 1
        
        while low < high:
            mid = (low + high + 1) // 2
            print(f"? {mid}\n")
            sys.stdout.flush()
            
            response = input().strip()
            
            if response == "Lucky!":
                low = mid
                balance += mid
            elif response == "Fraudster!":
                high = mid - 1
                balance -= mid
                if balance < 0:
                    return  # Terminate if fired
            elif response == "Fired!":
                return  # Terminate if fired
        
        # Output the determined value of M
        results.append(f"! {low}\n")
    
    # Print all results at the end
    for result in results:
        print(result)
        sys.stdout.flush()

File: exF.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_QUERIES 100000

void solve() {
    long long balance = 1; // Initial balance
    long long low = 0, high = 1e14, mid;
    char response[20];

    while (low < high) {
        mid = (low + high + 1) / 2;
        printf("? %lld\n", mid);
        fflush(stdout);

        scanf("%s", response);

        if (response[0] == 'L') { // "Lucky!"
            balance += mid;
            low = mid;
        } else if (response[0] == 'F') { // "Fraudster!"
            balance -= mid;
            high = mid - 1;
        } else if (response[0] == 'F' && response[1] == 'i') { // "Fired!"
            exit(0);
        }
    }

    printf("! %lld\n", low);
    fflush(stdout);
}

int main() {
    int t;
    scanf("%d", &t);

    while (t--) {
        solve();
    }

    return 0;
}